lib/src/sha512.dart

```diff
@@ -7,7 +7,7 @@ import 'dart:convert';
 import 'digest.dart';
 import 'hash.dart';
 // ignore: uri_does_not_exist
-import 'sha512_fastsinks.dart' if (dart.library.js) 'sha512_slowsinks.dart';
+import 'sha512_fastsinks.dart' if (dart.library.html) 'sha512_slowsinks.dart';
```

Comment:
`dart.library.js` exists in `libraries.json` for dart2wasm. It was used to support `allowInterop`, but that method has since then moved out of that library, so we should delete it from `libraries.json`.  You can't import it (you get a static error), but it exists.

Reply:
FWIW - the libraries.json format supports that middleground @lrhn is describing: where you can import a library unconditionally, but the `dart.library.xyz` returns false on conditional imports. This is true today for `dart.library.io` on the JS backends, for instance.

So if there is any need to still have a library, but denote it as excluded, we can do so with the same mechanism. (This is what ["supported: false"][1] is for)
[1]: https://github.com/dart-lang/sdk/blob/95cbc6a49724f9aaac6ba72fc69a14e9f9f53d70/sdk/lib/libraries.yaml#L269

That said, as mentioned in https://github.com/dart-lang/crypto/issues/180. I believe this specific case can be addressed for now using the `dart.library.ffi` condition.